entity/src/main/kotlin/com/mashup/dojo/PickRepository.kt

```diff
@@ -221,6 +228,80 @@ class PickRepositoryImpl(
             .limit(rank)
             .fetch()
     }
+
+    override fun findGroupByPickPaging(
+        pickedId: String,
+        sort: String,
+        pageable: Pageable,
+    ): Page<PickQuestionDetailMapper> {
+        val content = getGroupByPickContent(pickedId, sort, pageable)
+        val totalCount = getGroupByPickTotalCount(pickedId)
+
+        return PageImpl(content, pageable, totalCount)
+    }
+
+    private fun getGroupByPickContent(
+        pickedId: String,
+        sort: String,
+        pageable: Pageable,
+    ): List<PickQuestionDetailMapper> {
+        val pickEntity = QPickEntity.pickEntity
+        val questionEntity = QQuestionEntity.questionEntity
+        val imageEntity = QImageEntity.imageEntity
+
+        val query =
+            jpaQueryFactory
+                .select(
+                    QPickQuestionDetailMapper(
+                        pickEntity.id,
+                        questionEntity.id,
+                        questionEntity.content,
+                        imageEntity.url,
+                        // 가장 최근의 Pick 시간 가져오기
+                        pickEntity.createdAt.max().`as`("latestPickedAt"),
+                        pickEntity.id.count().`as`("totalReceivedPickCount")
+                    )
+                )
+                .from(pickEntity)
+                .join(questionEntity).on(pickEntity.questionId.eq(questionEntity.id))
+                .join(imageEntity).on(questionEntity.emojiImageId.eq(imageEntity.id))
+                .where(pickEntity.pickedId.eq(pickedId))
+                // Question, Image URL 기준으로 그룹화
+                .groupBy(pickEntity.questionId, imageEntity.url)
+
+        val sortedQuery = getSorted(sort, query)
+
+        return sortedQuery
+            .limit(pageable.pageSize.toLong())
+            .offset(pageable.offset)
+            .fetch()
+    }
+
+    private fun getSorted(
+        sort: String,
+        query: JPAQuery<PickQuestionDetailMapper>,
+    ): JPAQuery<PickQuestionDetailMapper> {
+        val pickEntity = QPickEntity.pickEntity
+        if (sort == PickSort.MOST_PICKED.name) {
+            return query.orderBy(Wildcard.count.desc(), pickEntity.createdAt.desc())
+        }
+
+        if (sort == PickSort.LATEST.name) {
+            return query.orderBy(pickEntity.createdAt.desc())
+        }
+
+        throw DojoException.of(DojoExceptionType.SORT_NOT_FOUND)
```

Comment:
```suggestion
        val pickEntity = QPickEntity.pickEntity

        return when (PickSort.findByValue(sort)) {
            PickSort.MOST_PICKED -> query.orderBy(Wildcard.count.desc(), pickEntity.createdAt.desc())
            PickSort.LATEST -> query.orderBy(pickEntity.createdAt.desc())
    }
```

요건 어떤가요 밑에 Picksort findByValue 정의해둔거 같던데
어차피 매핑 실패시에는 내부에서 throw 해주고 있고

Reply:
오호 역시 👍👍 또 배워갑니다.
근데 findByValue가 도메인에 있는 Enum에만 있어서, Repository에 있는 PickSort에도 findByValue 생성하겠습니다~!